### PR TITLE
Page animations

### PR DESCRIPTION
**Pull request by Wyvest**
## Checklist
- [ ] I made a clear description of what was changed
- [ ] I stated why these changes were necessary
- [ ] I updated documentation or said what needs to be updated
- [ ] I made sure these changes are backwards compatible
- [ ] This pull request is for one feature/bug fix

### CODE REVIEW

File: src/main/java/cc/polyfrost/oneconfig/gui/animations/Animation.java [hunk 1]
```diff
@@ -40,8 +40,9 @@ public abstract class Animation {
      * @param start    The start of the animation
      * @param end      The end of the animation
      * @param reverse  Reverse the animation
+     * @param b
      */
-    public Animation(float duration, float start, float end, boolean reverse) {
+    public Animation(float duration, float start, float end, boolean reverse, boolean b) {
```
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/gui/animations/Animation.java:45]: b

File: src/main/java/cc/polyfrost/oneconfig/gui/animations/ChainedAnimation.java [hunk 2]
```diff
@@ -27,17 +27,19 @@
 package cc.polyfrost.oneconfig.gui.animations;
 
 public class ChainedAnimation extends Animation {
-    private final Animation[] animations;
+    protected final Animation[] animations;
     private int currentAnimation = 0;
     private float value;
+    private float totalDuration = 0;
```
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/gui/animations/ChainedAnimation.java:33]: totalDuration is always 0

File: src/main/java/cc/polyfrost/oneconfig/gui/animations/quart/EaseInOutQuart.java [hunk 3]
```diff
@@ -35,11 +37,11 @@ public class EaseInOutQuart extends Animation {
      * @param reverse  Reverse the animation
      */
     public EaseInOutQuart(int duration, float start, float end, boolean reverse) {
-        super(duration, start, end, reverse);
+        super(duration, start, end, reverse, true);
     }
 
     @Override
-    protected float animate(float x) {
+    public float animate(float x) {
```
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/gui/animations/quart/EaseInOutQuart.java:44]: but why

**Review by DeDiamondPro — COMMENTED** [review]
some more things

File: src/main/java/cc/polyfrost/oneconfig/gui/animations/Animation.java (hunk 1 above)
Comment (DeDiamondPro) [reply src/main/java/cc/polyfrost/oneconfig/gui/animations/Animation.java:45]: (never used)

**Comment by Wyvest** [comment]
Closing until PolyUI is implemented.

**Pull request closed without merging**